iib/workers/tasks/build.py

```diff
@@ -589,22 +633,29 @@ def handle_add_request(
     """
     _verify_labels(bundles)
 
+    # Resolve bundles to their digests
+    resolved_bundles = _get_resolved_bundles(bundles)
```

Comment:
Could you consolidate `_verify_labels` and `_get_resolved_bundles` since you are essentially doing `skopeo inspect` on each bundle two times and that is a very expensive operation?

Reply:
Actually, it looks like `get_legacy_support_packages` calls skopeo inspect on every bundle a third time. Bonus points if you can consolidate that as well.

Reply:
I thought `_verify_labels` does a non-raw `skopeo inspect` while `_get_resolved_bundles` does a raw `skopeo inspect`. Hence, I did not think of combining the two. 

It does make sense to combine `_verify_labels` and `get_legacy_support_packages` though. But that is outside of the scope of this PR.

Please let me know if I am missing something?

Reply:
Ah you're right. Can you resolve the bundles before verifying the labels? That way the labels are verified on the same manifests that were selected.